custom-resource-handlers/src/certificate-signing-request.ts

```diff
@@ -0,0 +1,113 @@
+import aws = require('aws-sdk');
+import fs = require('fs');
+import os = require('os');
+import path = require('path');
+import util = require('util');
+
+import cfn = require('./_cloud-formation');
+import _exec = require('./_exec');
+import lambda = require('./_lambda');
+import _rmrf = require('./_rmrf');
+
+const secretsManager = new aws.SecretsManager();
+
+export async function main(event: cfn.Event, context: lambda.Context): Promise<void> {
+  try {
+    // tslint:disable-next-line:no-console
+    console.log(`Input event: ${JSON.stringify(event)}`);
+    const attributes = await handleEvent(event, context);
+    await cfn.sendResponse(event,
+                          cfn.Status.SUCCESS,
+                          event.LogicalResourceId || event.PhysicalResourceId || context.logStreamName,
+                          attributes);
+  } catch (e) {
+    // tslint:disable-next-line:no-console
+    console.error(e);
+    await cfn.sendResponse(event,
+                          cfn.Status.FAILED,
+                          event.LogicalResourceId,
+                          { CSR: '', SelfSignedCertificate: '' },
+                          e.message);
+  }
+}
+
+interface ResourceAttributes {
+  CSR: string;
+  SelfSignedCertificate: string;
+
+  [name: string]: string | undefined;
+}
+
+async function handleEvent(event: cfn.Event, _context: lambda.Context): Promise<ResourceAttributes> {
+  switch (event.RequestType) {
+  case cfn.RequestType.CREATE:
+  case cfn.RequestType.UPDATE:
+    return _createSelfSignedCertificate(event);
+  case cfn.RequestType.DELETE:
+    // Nothing to do - this is not a "Physical" resource
+    return { CSR: '', SelfSignedCertificate: '' };
+  }
+}
+
+async function _createSelfSignedCertificate(event: cfn.Event): Promise<ResourceAttributes> {
+  const tempDir = await util.promisify(fs.mkdtemp)(path.join(os.tmpdir(), 'x509CSR-'));
```

Comment:
A more common pattern is to invoke `promisify` immediately after the `import` statements and reuse the result:

```ts
import { promisify } from 'util';
import fs = require('fs')l;

const mkdtemp = promisify(fs.mkdtemp);
const readFile = promisify(fs.readFile);
// ...
```